frontend/src/features/books/BookDetails.jsx

```diff
@@ -30,13 +25,33 @@ function BookDetails() {
     fetchCurrentBook()
   }, [id])
 
-  if (!title) return <h3>Loading...</h3>
+  const deleteBook = async () => {
+    try {
+      const response = await fetch(`${API_URL}/${id}`, {
+        method: "DELETE"
+      })
+      if (response.ok) {
+        navigate('/')
+      } else {
+        const errorText = await response.text()
+        throw new Error(`Error ${response.status}: ${errorText}`)
+      }
+    } catch (e) {
+      console.log("An error occurred", e)
+    }
+  }
+
+  if (!book) return <h3>Loading...</h3>
 
   return (
     <div>
-      <h3>{title}</h3>
-      <p>Author: {author}</p>
-      <p>Description: {description}</p>
+      <h3>{book.title}</h3>
+      <p>Author: {book.author}</p>
+      <p>Description: {book.description}</p>
+      
+      <div className='post-links'>
+        <button onClick={deleteBook}>Delete</button>
+      </div>
       <Link to="/">Back to Books List</Link>
```

Comment:
@nshelokar-vf, please use proper naming conventions here

Reply:
Okay Sir. I will keep in mind.